### PR TITLE
fix: ci, .gitignore and rust toolchain updates, lint fixes

### PR DESCRIPTION
**Pull request by shuimuliang**

### CODE REVIEW

File: .github/workflows/lint-pr.yml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  main:
+    name: Lint
+    permissions:
+      pull-requests: read  # for amannn/action-semantic-pull-request to analyze PRs
+      statuses: write  # for amannn/action-semantic-pull-request to mark status of analyzed PR
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5.0.2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (nicbus) [inline .github/workflows/lint-pr.yml:23]: this workflow could add friction for contributors while not providing significant benefits, so let's drop this

File: .github/workflows/rust.yaml [hunk 2]
```diff
@@ -21,7 +21,7 @@ jobs:
         with:
           components: clippy
       - name: Lint
-        run: cargo clippy -- -D warnings
+        run: cargo clippy --workspace --all-targets --tests -- -D warnings
```
Comment (nicbus) [inline .github/workflows/rust.yaml:24]: as documented in the [github section](https://doc.rust-lang.org/stable/clippy/continuous_integration/github_actions.html) of the clippy documentation, let's use
```
cargo clippy --all-targets --all-features -- -D warnings
```

File: .gitignore [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
 # Test files
 /datacore
 /dataindex
 /dataldk0
 /dataldk1
 /dataldk2
 /tmp
+
+# will have compiled files and executables
+debug/
```
Comment (nicbus) [inline .gitignore:10]: this should not be needed

File: .gitignore [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
+/target
+
+# Remove Cargo.lock from gitignore if creating an executable, leave it for libraries
+# More information here https://doc.rust-lang.org/cargo/guide/cargo-toml-vs-cargo-lock.html
+Cargo.lock
```
Comment (nicbus) [inline .gitignore:15]: `Cargo.lock` is included on purpose, so builds use specific dependency versions (e.g. when running tests), please drop this and restore the file itself

File: .gitignore [hunk 5]
```diff
@@ -1,9 +1,29 @@
...
+
+# These are backup files generated by rustfmt
+**/*.rs.bk
+
+# MSVC Windows builds of rustc generate these, which store debugging information
+*.pdb
```
Comment (nicbus) [inline .gitignore:21]: windows is not supported so let's drop this

File: Cargo.toml [hunk 6]
```diff
@@ -33,7 +33,7 @@ lightning-net-tokio = { version = "0.0.115" }
 lightning-persister = { version = "0.0.115", path = "./rust-lightning/lightning-persister" }
 lightning-rapid-gossip-sync = { version = "0.0.115" }
 magic-crypt = "3.1.12"
-miniscript = { version = "8.0", features = ["serde"] }
+miniscript = { version = "9.0.1", features = ["serde"] }
```
Comment (nicbus) [inline Cargo.toml:36]: thanks for spotting this, let's keep the current version though, as dependency versions are usually updated before releasing a project update and after checking that all functionality works, also in related projects

File: rust-toolchain.toml [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[toolchain]
+channel = "nightly"
```
Comment (nicbus) [inline rust-toolchain.toml:2]: we prefer to build with the stable version of rust

**Review by nicbus — CHANGES_REQUESTED** [review]
Hi, thanks for providing these updates (I'll rename the PR to better reflect its contents).

Please see the requested changes. Also please squash the commits and sign them.

File: .github/workflows/rust.yaml (hunk 2 above)
Comment (shuimuliang) [reply .github/workflows/rust.yaml]: done.

File: .gitignore (hunk 4 above)
Comment (shuimuliang) [reply .gitignore]: done.

File: Cargo.toml (hunk 6 above)
Comment (shuimuliang) [reply Cargo.toml]: done.

File: rust-toolchain.toml (hunk 7 above)
Comment (shuimuliang) [reply rust-toolchain.toml]: done.

**Review by nicbus — CHANGES_REQUESTED** [review]
Please squash the commits and sign them.

**Comment by shuimuliang** [comment]
> Please squash the commits and sign them.

In fact, you could easily squash the commits when merging the pr.

**Review by nicbus — CHANGES_REQUESTED** [review]
@shuimuliang please:
- address the last unresolved conversation (restore the original `Cargo.lock` file)
- amend the commit and, while you're at it, update its message (dropping the description as it references changes that have been dropped)
- sign the amended commit

**Pull request closed without merging**